### PR TITLE
fix(flax): generate with logits processor/warper

### PR DESCRIPTION
**Pull request by borisdayma**
# What does this PR do?

This PR fixes an issue with flax generate method where the logits processor and warper were not being considered.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR. @patil-suraj @patrickvonplaten

### CODE REVIEW

File: src/transformers/generation_flax_utils.py [hunk 1]
```diff
@@ -560,10 +560,10 @@ def sample_search_body_fn(state):
 
             # apply min_length, ...
             logits = logits_processor(state.sequences, logits, state.cur_len)
-            # apply top_k, top_k, temperature
+            # apply top_p, top_k, temperature
             logits = logits_warper(logits, logits, state.cur_len)
 
-            next_token = jax.random.categorical(prng_key, model_outputs.logits[:, -1], axis=-1)
+            next_token = jax.random.categorical(prng_key, logits, axis=-1)
```
Comment (patil-suraj) [inline src/transformers/generation_flax_utils.py:566]:
```suggestion
            next_token = jax.random.categorical(prng_key, logits[:, -1], axis=-1)
```

**Review by patil-suraj — APPROVED** [review]
Great catch, thanks a lot for the fix!

**Review by patrickvonplaten — APPROVED** [review]
Thanks for fixing @borisdayma !
